packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart

```diff
@@ -325,15 +333,21 @@ class WebSocketConnection implements Closeable {
 
     try {
       _hasConnectionBroken = false;
-      _hubEventsController.add(SubscriptionHubEvent.connecting());
 
       await retryStrategy.retry(
         // Make a GET request
         () => _getPollRequest().timeout(_defaultRetryTimeout),
       );
+      _resetConnectionInit();
```

Comment:
Why is rest of init/variables needed here? Is this needed for new testing variables or something else? Calling _reconnectPending.complete() makes enough sense.

Reply:
The intent has been to start every new connection with a clean/blank slate. Nothing gets preserved except a list of existing subscriptions (so new listeners can be setup). Here, it's a little verbose by clearing the `_initMemo`, but the other completers need to be cleared for the new connection.